### PR TITLE
[develop] Add support to build bindings using mina as a submodule in o1js repo

### PR DESCRIPTION
**Pull request by MartinMinkov**
# Description
Addresses https://github.com/o1-labs/o1js/issues/1096

:link: Mina: https://github.com/MinaProtocol/mina/pull/14649
:link: o1js: https://github.com/o1-labs/o1js/pull/1318

This PR makes mina buildable as a submodule of o1js. To do so, the following changes have been made:
1. Moved `kimchi` into mina repo.
    - This was done for several reasons. Firstly, we want to replace the `snarkyjs` submodule in the mina repo with an o1js stub module with the same dependencies as o1js, but just checks compiling. This test ensures o1js can be compiled if anything changes in the mina repo. 
    - Secondly, there is a longer-term plan internally to create a o1 crypto repo that will combine the wasm bindings with the ocaml-rust stubs, and it's easier to colocate these libraries in the mina repo as of now.
2. Changes the dune files in a few places to accommodate the mina submodule in o1js
3. Changes the build scripts to accommodate the mina submodule in o1js
4. Removed codeowners entry for `kimchi`

### CODE REVIEW

**Review by mitschabaude — APPROVED** [review]

File: ocaml/dune [hunk 1]
```diff
@@ -51,7 +51,7 @@
  (modules snarky_js_types)
  (link_flags (-linkall))
  (modes native)
- (libraries mina_base fields_derivers_zkapps yojson)
```
Comment (dannywillems) [inline ocaml/dune:54]: I would have done this in a different commit than the one udpating the artefacts.

File: ocaml/lib/dune [hunk 2]
```diff
@@ -20,6 +20,7 @@
   pickles_base
   pickles.limb_vector
   pickles_types
+  key_cache
```
Comment (dannywillems) [inline ocaml/lib/dune:23]: Same

File: ocaml/lib/dune [hunk 3]
```diff
@@ -34,7 +35,7 @@
   data_hash_lib
   hash_prefixes
   fields_derivers
-  fields_derivers_zkapps
+  fields_derivers.zkapps
```
Comment (dannywillems) [inline ocaml/lib/dune:38]: Same

File: crypto/test_vectors/dump-test-vectors.sh [hunk 4]
```diff
@@ -3,13 +3,8 @@
 set -e
 
 pushd ../../../mina/src/lib/crypto/proof-systems/poseidon/export_test_vectors
-<<<<<<< HEAD
```
Comment (dannywillems) [inline crypto/test_vectors/dump-test-vectors.sh]: Same

File: README.md [hunk 5]
```diff
@@ -9,7 +9,6 @@ The repo is included as a git submodule in the [o1js repo](https://github.com/o1
 - `/compiled` - compiled JS and Wasm artifacts produced by `js_of_ocaml` and `wasm-bindgen` from Rust and OCaml source code. We keep these artifacts in the source tree so that developing on o1js can be done with standard JS tooling and doesn't require setting up the full OCaml/Rust build pipeline.
 - `/crypto` - pure TS implementations of a subset of the crypto primitives we use, including finite field and elliptic curve arithmetic. This is used by mina-signer (a pure TS package) to hash and sign transactions.
 - `/js` - JS-side wrappers for the artifacts located in `/compiled`, which differs between the Node.js and web versions of o1js. Includes code for setting up workers to support using `rayon` in Rust.
-- `/kimchi` - bindings to the [Kimchi proof system](https://o1-labs.github.io/proof-systems/kimchi/overview.html) which is implemented in Rust. This contains a Wasm compatibility layer written in Rust as well as a `js_of_ocaml`-to-`wasm-bindgen` glue layer written in JS.
```
Comment (dannywillems) [inline README.md:12]: More changes have to be made in this file I think, for instance the build instructions.
